### PR TITLE
Relax Frontmatter Parsing

### LINKED ISSUE #441: markdown parser should be more resilient
problems during markdown parsing should not throw a error and abort the pipeline.

eg:

```
Error: Found ambigous frontmatter fence: No empty line before the block! Make sure your frontmatter blocks contain no empty lines and your horizontal rules have an empty line before AND after them.
```

### PR DESCRIPTION
**Pull request by trieloff**
- no longer throw exceptions (we need to think about how to report this)
- do not require non-empty lines in the front matter block at the start of the document (same rules for other blocks)

fixes #441

### CODE REVIEW

File: src/html/parse-frontmatter.js [hunk 1]
```diff
@@ -301,7 +302,11 @@ const parseFrontmatter = ({ content = {} }) => {
         join('\n'),
       );
 
-      throw new FrontmatterParsingError(`${warning}\n${sourceref}`);
+      const err = new FrontmatterParsingError(`${warning}\n${sourceref}`);
```
Comment (koraa) [inline src/html/parse-frontmatter.js]: Why not just always warn?

File: src/html/parse-frontmatter.js [hunk 2]
```diff
@@ -173,7 +174,7 @@ const findFrontmatter = (mdast, str) => {
         'Found ambigous frontmatter fence: No empty line after the block! '
           + 'Make sure your frontmatter blocks contain no empty lines '
           + 'and your horizontal rules have an empty line before AND after them.');
-    } else if (src.match(re(`\\n${hspace}*\\n`))) {
+    } else if (src.match(re(`\\n${hspace}*\\n`)) && fst.idx > 0) {
```
Comment (koraa) [inline src/html/parse-frontmatter.js]: If you change this, you should also change the documentation appropriately.

File: src/html/parse-frontmatter.js (hunk 1 above)
Comment (trieloff) [reply src/html/parse-frontmatter.js]: Because you don't want to fill up the error logs in production with things that are not relevant at that time.

**Comment by koraa** [comment]
You could leave me a day to review…
Also did you address my feedback about using an option about relaxing rule 2?

**Comment by trieloff** [comment]
I'm trying to avoid options until they have proven a practical necessity.

**Pull request merged**